streams/src/main/java/org/apache/kafka/streams/state/internals/SegmentedBytesStore.java

```diff
@@ -114,7 +125,18 @@
          * @param to            ending time range
          * @return
          */
-        HasNextCondition hasNextCondition(final Bytes binaryKey, long from, long to);
+        HasNextCondition hasNextCondition(final Bytes binaryKey, final long from, final long to);
```

Comment:
my only concern with removing the single key method would be the potential performance impact of having to check both `<=` and `>=` for each value as opposed to just `=` for single key lookups.